vm/interpreter/src/default_runtime.rs

```diff
@@ -599,42 +606,28 @@ where
         &self.syscalls
     }
     fn total_fil_circ_supply(&self) -> Result<TokenAmount, ActorError> {
-        let get_actor_state = |addr: &Address| -> Result<ActorState, ActorError> {
-            self.state
-                .get_actor(&addr)
-                .map_err(|e| {
+        let total_circ = get_circulating_supply(
+            self.pre_ignition.as_ref().unwrap_or(
+                &setup_preignition_genesis_actors_testnet(&self.store).map_err(|e| {
                     e.downcast_default(
                         ExitCode::ErrIllegalState,
-                        "failed to get reward actor for cumputing total supply",
+                        "Failed to setup preignition genesis actors testnet",
                     )
-                })?
-                .ok_or_else(
-                    || actor_error!(ErrIllegalState; "Actor address ({}) does not exist", addr),
-                )
-        };
-
-        let rew = get_actor_state(&REWARD_ACTOR_ADDR)?;
-        let burnt = get_actor_state(&BURNT_FUNDS_ACTOR_ADDR)?;
-        let market = get_actor_state(&STORAGE_MARKET_ACTOR_ADDR)?;
-        let power = get_actor_state(&STORAGE_POWER_ACTOR_ADDR)?;
-
-        let st: power::State = self
-            .store
-            .get(&power.state)
-            .map_err(|e| {
-                e.downcast_default(
-                    ExitCode::ErrIllegalState,
-                    "failed to get storage power state",
-                )
-            })?
-            .ok_or_else(|| actor_error!(ErrIllegalState; "Failed to retrieve power state"))?;
-
-        let total = P::from_fil(P::TOTAL_FILECOIN)
-            - rew.balance
-            - market.balance
-            - burnt.balance
-            - st.total_pledge_collateral;
-        Ok(total)
+                })?,
+            ),
+            self.post_ignition.as_ref().unwrap_or(
+                &setup_postignition_genesis_actors_testnet(&self.store).map_err(|e| {
```

Comment:
so the difference here is that the go implementation only performs this computation once, where you are doing every single time total_fil_circ_supply is called. This cannot be this way